src/Control/Distributed/Process/Node.hs

```diff
@@ -389,7 +390,9 @@ forkProcess node proc =
                 (return . DiedException . (show :: SomeException -> String)))]
 
           -- [Unified: Table 4, rules termination and exiting]
-          modifyValidLocalState_ node (cleanupProcess pid)
+          mconns <- modifyValidLocalState node (cleanupProcess pid)
+          forM_ mconns (mapM_ NT.close)
```

Comment:
`forkIO` was meant to prevent the clean up from being interrupted in other places. In this occurrence, the clean up happens in a scope where exceptions are uninterruptibly masked, thus forkIO is not needed for that sake.

I do see that notifying the NC of the process death could delay. We could move closing connections after that, or use `forkIO` for uniformity as you are considering.

There is also the chance of threads trying to deliver exceptions to this one to block until the cleanup finishes. Another reason to use `forkIO`.